concept-wip/numbers/about.md

```diff
@@ -0,0 +1,245 @@
+# Numbers
+
+Crystal has a type `Number` which is the base type for all numeric types.
+Inside the Number type, there are two main types: [`Int`][int] and [`Float`][float].
+Int is short for integer and represents whole numbers.
+Int can be both negative and positive, and doesn't have a decimal point.
+
+```crystal
+1.class
+# => Int32
+
+-1.class
+# => Int32
+```
+
+Float represents numbers with a decimal point, Floats can be both negative and positive.
+
+```crystal
+10.5.class
+# => Float64
+
+1.0.class
+# => Float64
+```
+
+On numbers, you can use the [basic arithmetic operators: `+`, `-`, `*`, `/`, and `%`][math].
+
+## Converting between Int and Float
+
+You can convert between Int and Float using the `to_i` and `to_f` methods.
+When converting from Float to Int the number is rounded down.
+
+```crystal
+1.to_f
+# => 1.0
+
+1.0.to_i
+# => 1
+
+1.9.to_i
+# => 1
+```
+
+## Arithmetic operators
+
+You can use the basic arithmetic operators on Int and Float.
+The operators are `+`, `-`, `*`, `/`, and `%`.
+You can mix and match Int and Float when using these operators.
+
+### Addition & Subtraction
+
+Addition and subtraction are used to add or subtract numbers.
+The `+` operator is used for addition and the `-` operator is used for subtraction.
+When having an operation with a Float and an Int the result will be a Float.
+The result will be an Int if both numbers are Ints.
+
+```crystal
+1 + 1
+# => 2
+
+1.0 + 1
+# => 2.0
+
+4.5 - 1.2
+# => 3.3
+```
+
+### Multiplication & Division
+
+Multiplication is used to multiply numbers.
+The `*` operator is used for multiplication.
+When having an operation with a Float and an Int the result will be a Float.
+
+```crystal
+2 * 2
+# => 4
+
+2.0 * 2
+# => 4.0
+```
+
+Division is used to divide numbers.
+The `/` operator is used for division.
+The result will always be a Float.
+
+```crystal
+4 / 2
+# => 2.0
+
+4.0 / 2
+# => 2.0
+```
+
+````exercism/caution
+In some programming languages when dividing by zero the result will be an error.
+
+In Crystal when dividing by zero the result will be `Infinity` or `-Infinity`.
+The Only exception is when dividing zero by zero, which will result in `NaN` (Not a Number).
+Infinity and NaN are special values in the Float type.
+
+```crystal
+1 / 0
+# => Infinity
+
+-1 / 0
+# => -Infinity
+
+0 / 0
+# => NaN
+```
+````
+
+## Integer division
+
+Integer division is used to divide numbers and get the whole part of the result.
+The result will always be rounded down to an Int.
+
+```crystal
+4 // 2
+# => 2
+
+4.0 // 2
+# => 2
+
+4.5 // 2
+# => 2
+```
+
+````exercism/caution
+When dividing by zero when using integer division results in a `DivisionByZeroError`.
+This is different from normal division.
```

Comment:
I think this comment is for the whole markdown file?

I like to have a bit deeper explanation of species, and what I usually go over with myself is, how is this different from Ruby?

Since that is where most Crystal developers come from.
The crystal number system is based more on pythons than rubys (Not that they are super different but there are differences). I guess it is thereby I got a bit deeper.

I think some parts are there benefits of going a bit deeper on, and give a couple of examples. With the next update which I am going to push very soon. So has the doc been reduced in size. I am open to suggestions of further reductions to parts, but I think there are benefits to keeping some examples.